src/dotnet/commands/dotnet-compile-native/NativeCompileSettings.cs

```diff
@@ -168,8 +168,11 @@ private NativeCompileSettings()
             BuildType = DefaultBuiltType;
             NativeMode = DefaultNativeModel;
             AppDepSDKPath = Path.Combine(AppContext.BaseDirectory, "appdepsdk");            
-
-            _referencePaths = new List<string>(Directory.EnumerateFiles(AppDepSDKPath, "*.dll"));
+            _referencePaths = new Dictionary<string, string>();
+            foreach (var file in Directory.EnumerateFiles(AppDepSDKPath, "*.dll"))
+            {
+                _referencePaths.Add(Path.GetFileName(file), file);
```

Comment:
Is this being done to address a symlink issue? If not, then why do we need this since EnumerateFiles is expected to return an absolute path?